test/ts-scenario/steps/network-model.ts

```diff
@@ -1,34 +1,40 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
 /**
  * SPDX-License-Identifier: Apache-2.0
  */
 
 'use strict';
 
-import { Constants } from './constants';
+import {Constants} from './constants';
 import * as Gateway from './lib/gateway';
 import * as BaseUtils from './lib/utility/baseUtils';
-import { CommonConnectionProfileHelper } from './lib/utility/commonConnectionProfileHelper';
-import { StateStore } from './lib/utility/stateStore';
+import {CommonConnectionProfileHelper} from './lib/utility/commonConnectionProfileHelper';
+import {StateStore} from './lib/utility/stateStore';
 
-import { Given, Then, When } from 'cucumber';
+import {Given, Then, When} from 'cucumber';
 import * as path from 'path';
 
 const stateStore: StateStore = StateStore.getInstance();
 
-Given(/^I have a (.+?) backed gateway named (.+?) with discovery set to (.+?) for user (.+?) (?:in organization (.+?) )?using the connection profile named (.+?)$/, { timeout: Constants.STEP_MED as number }, async (walletType: string, gatewayName: string, useDiscovery: string, userName: string, orgName: string, ccpName: string) => {
+Given(/^I have a (.+?) backed gateway named (.+?) with discovery set to (.+?) for user (.+?) (?:in organization (.+?) )?using the connection profile named (.+?)$/, {timeout: Constants.STEP_MED as number}, async (walletType: string, gatewayName: string, useDiscovery: string, userName: string, orgName: string, ccpName: string) => {
 
 	const gateways: Map<string, any> = stateStore.get(Constants.GATEWAYS);
 	const fabricState: any = stateStore.get(Constants.FABRIC_STATE);
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-call
 	const tls: boolean = (fabricState.type.localeCompare('tls') === 0);
 
 	if (gateways && gateways.has(gatewayName)) {
 		BaseUtils.logMsg(`Gateway named ${gatewayName} already exists`);
 		if (walletType !== Constants.HSM_WALLET) {
 			return;
 		}
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
I think the logix is fixable here so disabling the rule is not required. Something like this might do it:
```
const gateway = gateways?.get(gatewayName);
if (gateway) {
    // all code in this if block
}
```

Reply:
gateway.gateway is taken as any.